include/roboteam_ai/control/controllers/PidTwoAxesController.h

```diff
@@ -0,0 +1,96 @@
+//
+// Created by ratoone on 25-09-19.
+//
+
+#ifndef RTT_PIDTWOAXESCONTROLLER_H
+#define RTT_PIDTWOAXESCONTROLLER_H
+
+#include <control/ControlUtils.h>
+#include "BaseController.h"
+#include "PidController.h"
+
+/**
+ * The PID class for 2 axes control (e.g. for 2D position or velocity control).
+ * Makes use of two instances of PidController
+ * @author Paul Vacariu
+ * @SInCE 25.09.2019
+ */
+class PidTwoAxesController : BaseController<Vector2,Vector2>{
+private:
+    //the 2 PIDs used in the controller: one for the X axis, and one the Y axis
+    PidController pidXAxis = PidController(0, 0, 0, 0);
+    PidController pidYAxis = PidController(0, 0, 0, 0);
+
+public:
+    /**
+     * Create a PID controller for two axes, with different parameters
+     * @param pidXParams a tuple with 4 parameters for X dir PID - proportional, integrative, derivative and feedforward gain
+     * @param pidYParams a tuple with 4 parameters for Y dir PID - proportional, integrative, derivative and feedforward gain
+     */
+    PidTwoAxesController(std::tuple<double, double, double, double> pidXParams, std::tuple<double, double, double, double> pidYParams);
+
+    /**
+     * Create a PID controller for two axes, with different parameters and zero feedforward
+     * @param pidXParams a tuple with 3 parameters for X dir PID - proportional, integrative, derivative
+     * @param pidYParams a tuple with 3 parameters for Y dir PID - proportional, integrative, derivative
+     */
+    PidTwoAxesController(std::tuple<double, double, double> pidXParams, std::tuple<double, double, double> pidYParams);
```

Comment:
It is also possible in C++ to use default arguments and then you can remove this overloaded constructor and use the previous constructor for this case